### ISSUE TITLE
[BE] 10-10EZ - Facilities: Pre-launch - Update Facilities API reference to V2

### ISSUE THREAD

**Issue by hdjustice** [department-of-veterans-affairs/va.gov-team#99214]
## Benefit to the Veteran
As a Veteran, I want to be presented the appropriate Facilities listing, so that I can easily find and select my preferred facility.

## Background
- We have inadvertently used a Facilities API that will soon be deprecated, and should migrate to V2.   

<details>
<Summary>Conversation with Facilities team</Summary>

> **Mitch Saltykov (he/him)**
>Hi, EZ/CG full-stack +dev here, and I'm a bit confused. We use the above client in both the Caregivers and EZ controllers (the latter is at `app/controllers/v0/health_care_applications_controller.rb:113`). But I thought it wasn't the Mobile one? They're both using `lib/lighthouse/facilities/v1/client.rb`
>
>**Eli Selkin**
>Yeah, that's the Mobile created one
>`/modules/facilities_api/app/services/facilities_api/v2/lighthouse/client.rb` our client is here
>
>**Mitch Saltykov (he/him)**
>was it intended to be Mobile-exclusive? Andy H and I built it out intending it to be a light update to v0 that both could use.
>
>**Eli Selkin**
>No, but they are going to migrate off it to ours
>Forgive us for all using the same name when referring to things... it's horrible for us as well
>
>**Mitch Saltykov (he/him)**
>So, is the upshot that EZ/CG should also migrate onto the v2 api? and (if I read the above right) that it should be a pretty smooth drop-in replacement?
>
>**Eli Selkin**
>I think it's pretty similar, since the LH V1 data is all the same (missing stuff we relied on in the past from their V0 but if you accepted the Mobile V1 then you're already knowledgeable about differences from v0)
>
>**Mitch Saltykov (he/him)**
>FWIW our usecases aren't too woeful -
>EZ needs all health facilities for a specified state
>(probably irrelevant to you, but we filter this down to active ones from a DB cache of what VES considers valid)
>CG does a proximity search to an address, and also pulls the parent facility with the Caregiver Support service if the selected one doesn't offer it.
>
>**Eli Selkin**
>Yeah, both cases (from a list of ids and area) are covered
>
>**Jill Adams**
>We don't have hard timing around deprecating the legacy one. We just would eventually love to maintain only 1, so it's not a fire currently, for planning purposes.

</details>

## Scope
- In-Scope
     - Updating the current Facilities API reference to V2
- Out-of-scope
     - Making any other changes within the Facilities pages

## Tasks
- [x] Use feature toggle to control the visibility for this feature
- [x] Update Facility search to use the new API

---
## Acceptance Criteria/Definition of Done
-  The Facilities search function will use the latest V2 API

**Comment by hdjustice** [comment]
Yeah, whatever makes the most sense.  So EZ should be exactly what we were thinking, and CG seems to be a bit more complicated?

**Comment by coope93** [comment]
@hdjustice That's what it looks like to me. I messaged the facilities team about it [here](https://dsva.slack.com/archives/C0FQSS30V/p1736890831755329?thread_ts=1734374386.293809&cid=C0FQSS30V) to make them aware we'll need some change to the client interface for our usage, but it shouldn't be too involved.

**Comment by hdjustice** [comment]
Ok, I will convert this one to EZ, and create another for CG then

**Comment by hdjustice** [comment]
#101118 created for CG work.

**Comment by hdjustice** [comment]
Validated that the facilities page is loading as expected, the state dropdown includes all states applicable to this search and the facility populates as expected when each new state is selected.  There does not appear to be any adverse effects due to this change.

**Comment by Mitch-A6** [comment]
enabled in production just now
